tests/periph_uart_mode/main.c

```diff
@@ -0,0 +1,181 @@
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Manual test application for UART mode on CPUs with only one UART
+ *
+ * @author      Ben Postman <ben.l.postman@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "periph/uart.h"
+#include "stdio_uart.h"
+#include "xtimer.h"
+
+/* Number of different options for each mode parameter */
+#define DATA_BIT_OPTIONS (4)
+#define PARITY_OPTIONS (5)
+#define STOP_BIT_OPTIONS (2)
+#define TOTAL_OPTIONS (DATA_BIT_OPTIONS * PARITY_OPTIONS * STOP_BIT_OPTIONS)
+
+/* Character positions in the mode string */
+#define DATA_BIT_POS (0)
+#define PARITY_POS (1)
+#define STOP_BIT_POS (2)
+
+/* Length of the printable mode string, includes null terminator */
+#define MODE_STR_LEN (4)
+
+/* UART_DEV is always 0 since this test is for device with 1 UART */
+#define _UART_DEV (UART_DEV(0))
+
+
+/* Delay between each mode, makes parsing results easier */
+#define DELAY_US 50000
+
+
+static void _get_mode(uart_data_bits_t data_bits, uart_parity_t parity,
+                        uart_stop_bits_t stop_bits, char* mode_str) {
+    switch (data_bits) {
+        case UART_DATA_BITS_5:
+            mode_str[DATA_BIT_POS] = '5';
+            break;
+        case UART_DATA_BITS_6:
+            mode_str[DATA_BIT_POS] = '6';
+            break;
+        case UART_DATA_BITS_7:
+            mode_str[DATA_BIT_POS] = '7';
+            break;
+        case UART_DATA_BITS_8:
+            mode_str[DATA_BIT_POS] = '8';
+            break;
+        default:
+            break;
+    }
+
+    switch (parity) {
+        case UART_PARITY_NONE:
+            mode_str[PARITY_POS] = 'N';
+            break;
+        case UART_PARITY_EVEN:
+            mode_str[PARITY_POS] = 'E';
+            break;
+        case UART_PARITY_ODD:
+            mode_str[PARITY_POS] = 'O';
+            break;
+        case UART_PARITY_MARK:
+            mode_str[PARITY_POS] = 'M';
+            break;
+        case UART_PARITY_SPACE:
+            mode_str[PARITY_POS] = 'S';
+            break;
+        default:
+            break;
+    }
+
+    switch (stop_bits) {
+        case UART_STOP_BITS_1:
+            mode_str[STOP_BIT_POS] = '1';
+            break;
+        case UART_STOP_BITS_2:
+            mode_str[STOP_BIT_POS] = '2';
+            break;
+        default:
+            break;
+    }
+
+    mode_str[MODE_STR_LEN - 1] = '\0';
+}
+
+
+int main(void)
+{
+    int8_t status;
+
+    /* Stores each mode string for printing at the end of the test */
+    char mode_strings[TOTAL_OPTIONS][MODE_STR_LEN];
+
+    /* Stores the result for each mode */
+    bool results[TOTAL_OPTIONS];
+
+    /* Results index that is used to associate mode_strings and results */
+    uint8_t ridx = 0;
+
+    /* Use arrays so that each option can be easily iterated over */
+    uart_data_bits_t data_bits[DATA_BIT_OPTIONS] = {
```

Comment:
You can make these const (I don't know if the compiler does that already though).

Reply:
Done